AstSemantics.md

```diff
@@ -4,6 +4,11 @@ WebAssembly code is represented as an Abstract Syntax Tree (AST) where each node
 represents an expression. Each function body consists of exactly one expression. 
 All expressions and operators are typed, with no implicit conversions or overloading rules.
 
+This document explains the high-level design of the AST: its types, constructs, and
+semantics. For implementation details consult [the formal spec](https://github.com/WebAssembly/spec),
```

Comment:
You're arguing @kripken's words here, I'll pick what you suggest but @kripken let me know if you'd like something else as well.